authentication/subscription/service/oso_subscription_service.go

```diff
@@ -170,10 +170,50 @@ func (s *osoSubscriptionServiceImpl) loadSubscriptions(ctx context.Context, user
 		log.Error(ctx, map[string]interface{}{
 			"err":         err,
 			"reg_app_url": regAppURL,
+			"username":    username,
 			"body":        bodyString,
 		}, "unable to unmarshal json with subscription status")
 		return nil, autherrors.NewInternalError(ctx, err)
 	}
 
 	return &sbs, nil
 }
+
+// DeactivateUser deactivates the user on OpenShift
+func (s *osoSubscriptionServiceImpl) DeactivateUser(ctx context.Context, username string) error {
+	// Load status from OSO
+	regAppURL := fmt.Sprintf("%s/api/accounts/%s/deprovision_osio?authorization_username=%s",
+		s.config.GetOSORegistrationAppURL(), username, s.config.GetOSORegistrationAppAdminUsername())
+	log.Debug(ctx, map[string]interface{}{
+		"url": regAppURL,
+	}, "calling remote registration application to check the user status")
+	req, err := http.NewRequest("GET", regAppURL, nil)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":         err.Error(),
+			"reg_app_url": regAppURL,
+		}, "unable to create http request")
+		return errs.Wrapf(err, "unable to deprovision user")
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", s.config.GetOSORegistrationAppAdminToken()))
+	res, err := s.httpClient.Do(req)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":         err.Error(),
+			"reg_app_url": regAppURL,
+		}, "unable to deprovision user")
+		return errs.Wrapf(err, "unable to deprovision user")
+	}
+	defer rest.CloseResponse(res)
+	bodyString := rest.ReadBody(res.Body)
+
+	if res.StatusCode != http.StatusOK {
```

Comment:
@nurali-techie could you confirm that the expected status is `200`? Not `204` or another `20x`?

Reply:
successful status is `HTTP 200` only :+1:

Reply:
@xcoulon we also need to handle `HTTP 404`, this happens when for given user there is NO osio_subscription exists.  Previously, we (me + @alexeykazakov) decided that it's NOT error state, the same should be reflected here.

Reply:
@nurali-techie ok, so in case the remote reg app service returns a `404`, this service here will not report an error.

Reply:
done in d770717